src/javascript/DesignSystem/Input/Input.jsx

```diff
@@ -0,0 +1,135 @@
+        root: {
+            borderRadius: '1px',
+            background: theme.palette.field.alpha,
+            border: `1px solid ${theme.palette.ui.omega}`,
+            boxSizing: 'border-box',
+            '&:hover:not($disabled):not($focused):not($error):not($readOnly)': {
+                border: `1px solid ${theme.palette.ui.zeta}`
+            },
+            fontSize: theme.typography.iota.fontSize,
+            transitionDuration: '.3s',
+            padding: '3px 12px 3px 12px'
+        },
+        focused: {
+            border: `1px solid ${theme.palette.brand.alpha}`
+        },
+        readOnly: {
+            background: theme.palette.ui.epsilon,
+            border: `1px solid ${theme.palette.ui.omega}`
+        },
+        // Hack for disabled style as the default disabled style applies on both container AND input element.
+        inputDisabled: {
+            background: theme.palette.ui.epsilon,
+            border: `1px solid ${theme.palette.ui.zeta}`,
+            color: theme.palette.font.gamma
+        },
+        error: {
+            border: `1px solid ${theme.palette.support.alpha}`
+        },
+        inputAdornedStart: {
+            transitionDuration: '.3s'
+        },
+        inputAdornedStartFocus: {
+            color: theme.palette.brand.alpha
+        },
+        inputAdornedStartError: {
+            color: theme.palette.support.alpha
+        }
+    };
+};
+
+export const Input = ({classes, defaultValue, disabled, error, fullWidth, id, variant, inputProps, name, onBlur, onChange, onFocus, readOnly, value, ...others}) => {
+    const [focus, setFocus] = useState(false);
+    const handleFocus = () => {
+        onFocus();
+        setFocus(true);
+    };
+
+    const handleBlur = () => {
+        onBlur();
+        setFocus(false);
+    };
+
+    const {icon, interactive} = variant;
+
+    const {readOnly: readOnlyClass, inputDisabled, inputAdornedStart, inputAdornedStartFocus, inputAdornedStartError, ...containerClasses} = classes;
+    return (
+        <InputBase
+            id={id}
+            defaultValue={defaultValue}
+            name={name}
+            classes={containerClasses}
+            className={`${readOnly ? readOnlyClass : ''} ${disabled ? inputDisabled : ''}`}
+            disabled={disabled}
+            error={error}
+            fullWidth={fullWidth}
+            inputProps={inputProps}
+            readOnly={readOnly}
+            startAdornment={icon &&
+            <InputAdornment
+                className={`${inputAdornedStart} ${!readOnly && focus ? inputAdornedStartFocus : ''} ${!readOnly && error ? inputAdornedStartError : ''}`}
+                position="start"
+            >
+                {icon}
+            </InputAdornment>}
+            endAdornment={interactive && <InputAdornment position="end">{interactive}</InputAdornment>}
+            value={value}
+            onBlur={handleBlur}
+            onFocus={handleFocus}
+            onChange={onChange}
```

Comment:
Tu ne fais aucun traitement dessus, autant le `{...other}` gérer ces props

Reply:
onBlur and onFocus are used by the focus state handling of the component.
As we want to expose our component (Material UI should be hidden) we need define onChange and value as prop used by some of our components.

Reply:
They have to be in the prop-types for sure, but what is the gain to do ` onChange={onChange}` vs `{...other}` ?